server/src/instant/model/schema.clj

```diff
@@ -141,50 +141,171 @@
-       "We cannot automatically swap the direction of the link, so you will need to do this manually. "
-       "Check your schema in https://www.instantdb.com/dash for a link with the same label names. "
-       "If you find one, you can either delete the existing link in the dashboard, or swap the `forward` and `reverse` parameters for this link in your schema file."))
+       "We cannot automatically swap the direction of the link. "
+       "To fix this, can: a) swap the `forward` and `reverse` parameters for this link in your schema file, or b) delete the existing link in the dashboard."
+       "Check your full schema in the dashboard for a link with the same label names: "
+       "https://www.instantdb.com/dash?s=main&t=explorer"))
 
 (defn assert-unique-idents! [current-attrs steps]
-  (let [current-fwd-ident-names (->> current-attrs
-                                     (map attr-model/fwd-ident-name)
-                                     (map vec))
-        current-all-ident-names (->> current-attrs
-                                     (mapcat attr-ident-names)
-                                     (map vec))
-        new-ident-names (->>
-                         steps
-                         (mapcat (fn [[op data]]
-                                   (when (= op :add-attr)
-                                     (attr-ident-names data)))))
-        dups (->> (concat current-all-ident-names new-ident-names)
-                  (frequencies)
-                  (filter (fn [[_ freq]] (> freq 1))))
-        errors (map (fn [[[etype label]]]
-                      {:in [:schema]
-                       :message (if
-                                 (some #(= % [etype label]) current-fwd-ident-names)
-                                  (dup-message etype label)
-                                  (backwards-link-message etype label))})
-                    dups)]
-    (ex/assert-valid! :schema
-                      :steps
-                      errors)))
+  (let [current-link-attrs
+        (filter (comp #{:ref} :value-type) current-attrs)
+
+        current-blobs
+        (filter (comp #{:blob} :value-type) current-attrs)
+
+        current-blob-idents
+        (->> current-blobs
+             (map #(attr-model/fwd-ident-name %))
+             (into #{}))
+
+        current-links-mapping-fwd
+        (->> current-link-attrs
+             (map #(vector
+                    (attr-model/fwd-ident-name %)
+                    (attr-model/rev-ident-name %)))
+             (into {}))
+
+        current-links-mapping-rev
+        (->> current-link-attrs
+             (map #(vector
+                    (attr-model/rev-ident-name %)
+                    (attr-model/fwd-ident-name %)))
+             (into {}))
+
+        errors
+        (->>
+         steps
+         (filter (comp #{:add-attr} first))
+         (map second)
+         (map #(let [fwd-name
```

Comment:
nit: would extract this into a helper function. 

Reason: later on when debugging, it can be very helpful to add a tool/def-locals here, but right now it will be hard because of the %.

Reply:
Should I do a helper, or just change the `#` shorthand to `fn`?

Reply:
I would probably do a helper, as it's pretty big